core/capabilities/integration_tests/streams_test.go

```diff
@@ -0,0 +1,91 @@
+package integration_tests
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/feeds_consumer"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+)
+
+var (
+	workflowName    = "abcdef0123"
+	workflowOwnerID = "0100000000000000000000000000000000000001"
+)
+
+func Test_AllAtOnceTransmissionSchedule(t *testing.T) {
+	ctx := testutils.Context(t)
+
+	// The don IDs set in the below calls are inferred from the order in which the dons are added to the capabilities registry
+	// in the setupCapabilitiesRegistryContract function, should this order change the don IDs will need updating.
+	workflowDonInfo := createDonInfo(t, don{id: 1, numNodes: 5, f: 1})
+	triggerDonInfo := createDonInfo(t, don{id: 2, numNodes: 7, f: 1})
+	targetDonInfo := createDonInfo(t, don{id: 3, numNodes: 4, f: 1})
+
+	consumer, feedIDs, triggerSink := setupStreamDonsWithTransmissionSchedule(ctx, t, workflowDonInfo, triggerDonInfo, targetDonInfo, 3,
+		"2s", "allAtOnce")
+
+	reports := []*datastreams.FeedReport{
+		createFeedReport(t, big.NewInt(1), 5, feedIDs[0], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(3), 7, feedIDs[1], triggerDonInfo.keyBundles),
+		createFeedReport(t, big.NewInt(2), 6, feedIDs[2], triggerDonInfo.keyBundles),
+	}
+
+	triggerSink.sendReports(reports)
+
+	waitForConsumerReports(t, consumer, reports)
```

Comment:
This is an assertion function, so could probalby stay in the test file :man_shrugging: